File: src/test/java/org/gradle/plugins/fsm/util/BaseConfiguration.java
package org.gradle.plugins.fsm.util;

import de.espirit.firstspirit.module.Configuration;
import de.espirit.firstspirit.module.ServerEnvironment;

import javax.swing.*;
import java.awt.*;
import java.util.Set;

public class BaseConfiguration implements Configuration<ServerEnvironment> {
    @Override
    public boolean hasGui() {
        return false;
    }

    @Override
    public JComponent getGui(Frame frame) {
        return null;
    }

    @Override
    public void load() {

    }

    @Override
    public void store() {

    }

    @Override
    public Set<String> getParameterNames() {
        return null;
    }

    @Override
    public String getParameter(String s) {
        return null;
    }

    @Override
    public void init(String s, String s1, ServerEnvironment t) {

    }

    @Override
    public ServerEnvironment getEnvironment() {
        return null;
    }
}

File: src/test/kotlin/org/gradle/plugins/fsm/util/BaseConfiguration.kt
package org.gradle.plugins.fsm.util

import de.espirit.firstspirit.module.Configuration
import de.espirit.firstspirit.module.ServerEnvironment
import java.awt.Frame
import javax.swing.JComponent

open class BaseConfiguration: Configuration<ServerEnvironment> {

    override fun hasGui(): Boolean {
        return false
    }

    override fun getGui(frame: Frame?): JComponent? {
        return null
    }

    override fun load() {}

    override fun store() {}

    override fun getParameterNames(): Set<String?>? {
        return null
    }

    override fun getParameter(s: String?): String? {
        return null
    }

    override fun init(s: String?, s1: String?, t: ServerEnvironment?) {}

    override fun getEnvironment(): ServerEnvironment? {
        return null
    }

}